### PR TITLE
FIX: Clear unread when channel archived

### PR DESCRIPTION
**Pull request by martin-brennan**
When the channel was archived some users could end
up with an unclearable unread indicator. This commit
fixes the issue but for now the user still needs to
reload for the channel to drop off the list.

### CODE REVIEW

File: lib/chat_channel_archive_service.rb [hunk 1]
```diff
@@ -225,7 +226,9 @@ def notify_archiver(result, error: nil)
     )
   end
 
-  def kick_all_users
-    UserChatChannelMembership.where(chat_channel: chat_channel).update_all(following: false)
+  def kick_all_users(last_message_id)
```
Comment (tgxworld) [inline lib/chat_channel_archive_service.rb:229]: Just wonder why we need to pass this as an argument when this method can derive the `last_message_id` from the `chat_channel` object.

File: spec/lib/chat_channel_archive_service_spec.rb [hunk 2]
```diff
@@ -127,16 +127,29 @@ def start_archive
...
 
-      it "unfollows (leaves) the channel for all users" do
-        create_messages(3)
-        channel.chat_messages.map(&:user).each do |user|
-          UserChatChannelMembership.create(chat_channel: channel, user: user, following: true)
+      describe "channel members" do
+        before do
+          create_messages(3)
+          channel.chat_messages.map(&:user).each do |user|
+            UserChatChannelMembership.create(chat_channel: channel, user: user, following: true)
```
Comment (tgxworld) [inline spec/lib/chat_channel_archive_service_spec.rb:134]:
```suggestion
            UserChatChannelMembership.create!(chat_channel: channel, user: user, following: true)
```

File: assets/javascripts/discourse/services/chat.js [hunk 3]
```diff
@@ -560,6 +561,20 @@ export default Service.extend({
         }
 
         channel.set("status", busData.status);
+
+        // it is not possible for the user to set their last read message id
+        // if the channel has been archived, because all the messages have
+        // been deleted. we don't want them seeing the blue dot anymore so
+        // just completely reset the unreads
+        if (busData.status === CHANNEL_STATUSES.archived) {
```
Comment (tgxworld) [inline assets/javascripts/discourse/services/chat.js:569]: Will we be able to test for this?

**Review by tgxworld — APPROVED** [review]
Just a couple of comments but the approach is OK

File: lib/chat_channel_archive_service.rb (hunk 1 above)
Comment (martin-brennan) [reply lib/chat_channel_archive_service.rb:229]: good point will change

File: assets/javascripts/discourse/services/chat.js (hunk 3 above)
Comment (martin-brennan) [reply assets/javascripts/discourse/services/chat.js:569]: Hmm I think so, we can send MessageBus messages in JS tests

Comment (martin-brennan) [reply assets/javascripts/discourse/services/chat.js:569]: Will add one

**Pull request merged**